src/Reader.cc

```diff
@@ -66,33 +105,41 @@ class ReaderNewInstanceWorker : public Napi::AsyncWorker {
 
     pulsar_result result =
         pulsar_client_create_reader(this->cClient, topic.c_str(), this->readerConfig->GetCStartMessageId(),
-                                    this->readerConfig->GetCReaderConfig(), &(this->cReader));
-    delete this->readerConfig;
+                                    this->readerConfig->GetCReaderConfig(), &this->readerWrapper->cReader);
     if (result != pulsar_result_Ok) {
       SetError(std::string("Failed to create reader: ") + pulsar_result_str(result));
       return;
+    } else {
+      this->listener = this->readerConfig->GetListenerCallback();
     }
+    delete this->readerConfig;
```

Comment:
if `result` is not ok, won't this line be executed?

Reply:
Exactly. If the result is not ok, this is not be executed.
I fixed that this is be done if the result is not ok.

Reply:
I think it is better to write as follows.
https://github.com/apache/pulsar-client-node/blob/v1.3.0/src/Consumer.cc#L180-L187

Reply:
I implemented `createReaderCallback` like [subscribeCallback](https://github.com/apache/pulsar-client-node/blob/v1.3.0/src/Consumer.cc#L178-L190).